### PR TITLE
Update Helm chart to install Crossplane images from xpkg.upbound.io

### LINKED ISSUE #3560: Publish Crossplane image to xpkg.upbound.io
### What problem are you facing?

A few months ago, many of the packages in the Crossplane ecosystem moved from Dockerhub to `xpkg.upbound.io` due to a few motivating factors:
- Rate limiting on pulls from Dockerhub.
- Automatic documentation support for all API types in a package on marketplace.upbound.io.
- Ability to search and discover packages based on their metadata on marketplace.upbound.io.

Crossplane itself is published as an OCI image and Helm chart, with the latter referencing the former. This differs from the structure of Crossplane packages, but there are a few commonalities, namely:
- Crossplane bundles its API types in its OCI image.
- Pulling the Crossplane image is still subsequent to rate limiting as imposed by whatever registry it is hosted on.

### How could Crossplane help solve your problem?

Moving the Crossplane image to be hosted on xpkg.upbound.io would ensure that the community is not negatively impacted by any rate limiting, and would create the opportunity for the Crossplane API documentation to live alongside Crossplane packages (and similarly be automatically updated when new versions are published).

Moving the image would likely have very minimal impact on existing users, as users install Crossplane by downloading the Helm chart, which would be updated to reference the new location by default. There are two cases where a user could potentially have difficulty in upgrading:
- They have Dockerhub on an allow list in their cluster and do not have xpkg.upbound.io on the list.
- They are automatically syncing Crossplane versions from Dockerhub to an internal registry and then accessing the internal registry from their cluster.

Both of these issues could be mitigated by proper community messaging and a reasonable transition window. I would propose that we dual publish to Dockerhub and xpkg.upbound.io and update to point at xpkg.upbound.io in the Crossplane helm chart by default for the next release (`v1.11.0`). This would be accompanied by messaging in release notes and Slack prior to the release. At the next release (`v1.12.0`), Dockerhub could be dropped as an official host. This would give a reasonable window for users that need to update any internal systems to pull from xpkg.upbound.io a full release cycle to do so (they would be required to upgrade with `docker.io` specified in the values file for the mean time), while going ahead and getting the vast majority of the community over to xpkg.upbound.io on `v1.11.0`.

**Amendment**

Instead of moving the default in the Helm chart in `v1.11.0`, we will add a note in the `v1.11.0` release notes that the default will be changing in `v1.12.0`, then stop publishing to Dockerhub and update the default as part of the `v1.12.0` release.

### PR DESCRIPTION
**Pull request by jbw976**
### Description of your changes

This PR simply updates Crossplane's Helm chart to install the Crossplane image from `xpkg.upbound.io`.

Fixes #3560

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] ~Added or updated unit tests.~
- [ ] ~Added or updated e2e tests, if necessary. (check with reviewers/maintainers if you're unsure whether E2E tests are necessary for the change).~
- [x] Run `make reviewable` to ensure this PR is ready for review.
- [ ] ~Added `backport release-x.y` labels to auto-backport this PR, if necessary.~
- [x] Opened a PR updating the [docs], if necessary.
  - https://github.com/crossplane/docs/pull/572

### How has this PR been tested?

I tested these changes locally in a kind cluster by installing the local Helm chart with a `helm` client and setting the most recent version of the Crossplane image that has been published to `xpkg.upbound.io`: `v1.14.0-rc.0.611.g8121e6d6`:

```
❯ .cache/tools/darwin_arm64/helm-v3.13.1 install crossplane --namespace crossplane-system --create-namespace cluster/charts/crossplane --set image.tag=v1.14.0-rc.0.611.g8121e6d6 --set 'args={"--debug"}'
```

Note this doesn't set `image.repository`, so the new value of `xpkg.upbound.io/crossplane/crossplane` in the chart that is updated in this PR is used.

The Crossplane pods get to `Running` successfully:

```
❯ kcs get pod
NAME                                       READY   STATUS    RESTARTS   AGE
crossplane-7dd756b54d-jpxql                1/1     Running   0          29s
crossplane-rbac-manager-765d64bb64-wp4gw   1/1     Running   0          29s
```

And we can see that both pods are using the expected container image from `xpkg.upbound.io`:

```
❯ kcs get pod -o json | jq '.items[] | .metadata.name + " " + .spec.containers[].image'
"crossplane-7dd756b54d-jpxql xpkg.upbound.io/crossplane/crossplane:v1.14.0-rc.0.611.g8121e6d6"
"crossplane-rbac-manager-765d64bb64-wp4gw xpkg.upbound.io/crossplane/crossplane:v1.14.0-rc.0.611.g8121e6d6"
```

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

### CODE REVIEW

File: cluster/charts/crossplane/README.md.gotmpl [hunk 1]
```diff
@@ -93,7 +93,7 @@ replicas: 1
 deploymentStrategy: RollingUpdate
 
 image:
-  repository: crossplane/crossplane
+  repository: xpkg.upbound.io/crossplane/crossplane
```
Comment (jbw976) [inline cluster/charts/crossplane/README.md.gotmpl:96]: i'm not actually sure what this `README.md.gotmpl` file is used for or if there is any automation around it, any ideas?

File: cluster/charts/crossplane/README.md [hunk 2]
```diff
@@ -78,7 +78,7 @@ and their default values.
 | `extraVolumesCrossplane` | Add custom `volumes` to the Crossplane pod. | `{}` |
 | `hostNetwork` | Enable `hostNetwork` for the Crossplane deployment. Caution: enabling `hostNetwork`` grants the Crossplane Pod access to the host network namespace. | `false` |
 | `image.pullPolicy` | The image pull policy used for Crossplane and RBAC Manager pods. | `"IfNotPresent"` |
-| `image.repository` | Repository for the Crossplane pod image. | `"crossplane/crossplane"` |
+| `image.repository` | Repository for the Crossplane pod image. | `"xpkg.upbound.io/crossplane/crossplane"` |
```
Comment (jbw976) [inline cluster/charts/crossplane/README.md:81]: @plumbis I think I'll need to open a docs PR to update https://github.com/crossplane/docs/blob/master/content/master/software/install.md - there's no automation keeping that file in sync with this README.md, is there?

File: cluster/charts/crossplane/README.md.gotmpl (hunk 1 above)
Comment (phisco) [reply cluster/charts/crossplane/README.md.gotmpl:96]: Yes, it's automatically rendered to the actual README of the chart with all the values updated, this part feels a bit superfluous though, we could remove it

**Review by phisco — APPROVED** [review]

**Review by turkenh — APPROVED** [review]

File: cluster/charts/crossplane/README.md (hunk 2 above)
Comment (jbw976) [reply cluster/charts/crossplane/README.md:81]: @plumbis confirmed that a separate manual PR is needed 🤓

**Pull request merged**